pyro/poutine/trace_struct.py

```diff
@@ -124,12 +127,18 @@ def log_prob_sum(self, site_filter=lambda name, site: True):
         result = 0.0
         for name, site in self.nodes.items():
             if site["type"] == "sample" and site_filter(name, site):
-                try:
+                if "log_prob_sum" in site:
                     log_p = site["log_prob_sum"]
-                except KeyError:
-                    log_p = site["fn"].log_prob(site["value"], *site["args"], **site["kwargs"])
-                    log_p = scale_and_mask(log_p, site["scale"], site["mask"]).sum()
-                    site["log_prob_sum"] = log_p
+                else:
+                    try:
+                        log_p = site["fn"].log_prob(site["value"], *site["args"], **site["kwargs"])
+                        log_p = scale_and_mask(log_p, site["scale"], site["mask"]).sum()
+                        site["log_prob_sum"] = log_p
+                    except ValueError:
```

Comment:
I don't think we need a `try..except ValueError` here again, as the error would be correctly logged by `.log_prob` downstream.

Reply:
You are right, I missed that. I will fix it